src/pages/iv-items/index.js

```diff
@@ -0,0 +1,181 @@
+import { useContext } from 'react'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+
+import { useWindow } from 'src/windows'
+import { ControlContext } from 'src/providers/ControlContext'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import ItemWindow from './window/ItemWindow'
+
+const IvItems = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    const response = await getRequest({
+      extension: InventoryRepository.Items.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_params=&filter=&_sortField=`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    access,
+    search,
+    clear,
+    invalidate
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: InventoryRepository.Items.qry,
+    datasetId: ResourceIds.Items,
+    search: {
+      endpointId: InventoryRepository.Items.snapshot,
+      searchFn: fetchWithSearch
+    }
+  })
+
+  async function fetchWithSearch({ qry }) {
+    const response = await getRequest({
+      extension: InventoryRepository.Items.snapshot,
+      parameters: `_filter=${qry}&_startAt=0&_size=50`
```

Comment:
isn't it better to use this way instead of writing startat and pagesize hardcoded;
moreover it is pagesize not size?

async function fetchWithSearch({ options = {}, qry }) {
    const { _startAt = 0, _pageSize = 50 } = options

    return await getRequest({
      extension: GeneralLedgerRepository.ChartOfAccounts.snapshot,
      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_filter=${qry}`
    })
  }

Reply:
yes